### DISCUSSION TITLE
Comparison to stripe-rs

### DISCUSSION THREAD

**Discussion by seanpianka** [arlyon/async-stripe#95] [General]
Hi @arlyon, thanks for making this fork!

Besides the async support, could you describe how else async-stripe differs from stripe-rs? Is there support for other API features that are currently missing from the codegen in the stripe-rs project?

Namely, support for:
* idempotent requests, connection pooling, retries
* 2020 API
* Checkout,
* webhooks (also, there's a  [security vulnerability](https://github.com/wyyerd/stripe-rs/issues/165) in the upstream crate)

I'd be willing to help port over any missing functionality that's in stripe-rs and not available in this project yet. Hopefully, Stripe is start supporting an official Rust crate soon...

**Comment by arlyon** [comment]
Hey Sean!

I have a post here in the discussions that lists changes. https://github.com/arlyon/async-stripe/discussions/79

While I originally was planning on upstreaming (which is why I was pushing to get us maintainer access), this diverged fairly significantly in the interim and I don't feel comfortable essentially 'overwriting' the other project without support (or interest) from the original maintainers. That said, if they are on board I am very happy to merge all these changes back in. Though I will say, as time went on I became more and more opinionated on this fork.

To answer your bullets:

- no work has been done on idempotency, though I'd like to see it included at some point. it's just a question of finding the time
- the api is regenerated weekly and should be mostly up to date (though there is an issue with the openapi definitions from the last 2 months or so that causes builds to fail; there are some types that aren't generated correctly)
- I added some codegen changes that I believe fixed the checkout api (see this issue https://github.com/arlyon/async-stripe/issues/24)
- no work has been done on webhooks

**Comment by seanpianka** [comment]
Thanks for the quick response! 

I'd bet the upstream maintainers would be open to upstreaming each of the items in #79. They're all major improvements compared to what stripe-rs has on `master` or (especially) crates.io currently. I'm pushing for this effort to upstream solely because (I estimate that) it has more active users presently, and all this good work would get into the hands of a greater number of people faster. 

My estimation might be off of course (and the recent downloads on crates.io do tell a different story), but I predict there's a decent group of users currently pulling it via a git dependency. The larger the community, the better? :thinking:

**Comment by arlyon** [comment]
(converted this into a discussion) If we can get the green light from the core maintainers then lets go for it: the sooner the better really!
